### ISSUE TITLE
Web -Payment - Popup with payment option is hidden title for new public accounts

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11414]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify.com
2. Log in with your band new public account
3. Go to Settings - Payments
4. Click Add Payments

## Expected Result:
Popup with payment option is not hidden title for new public accounts

## Actual Result:
Popup with payment option is hidden title for new public accounts

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.9.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any public accounts

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/193019953-2b14be80-bdc3-40d6-8cd8-4fc83f09e99d.mp4

![Payment](https://user-images.githubusercontent.com/93399543/193019977-28c830b9-e126-4302-a005-1131f378c522.png)

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by stephanieelliott** [comment]
Posted to Upwork: https://www.upwork.com/jobs/~01f78d3df2e9e52ada

**Comment by aldo-expensify** [comment]
> Why was the popover made to open from the top? IIRC, the popover used to open from the bottom of the button.

I guess it could be because there are views where this button is at the bottom of the screen, like this:

<img src="https://user-images.githubusercontent.com/87341702/194188593-e99170b2-ea7c-45ce-8267-16ad09907710.png" width="300">

Maybe we should add a prop to control if it appears on top or on bottom?

**Comment by tienifr** [comment]
# Proposal

## Problem
In the Payment Page, We are leaving the default menu option displayed above the button

## Solution
we should add a prop to control if it appears on top or on the bottom as this comment: https://github.com/Expensify/App/issues/11414#issuecomment-1269153706

Link PR: https://github.com/tienifr/App/pull/6
Result: 
<img width="1440" alt="Screen Shot 2022-10-07 at 11 36 10" src="https://user-images.githubusercontent.com/113963320/194473319-623acf8d-10c7-4ea3-92c3-35c8756cb432.png">

**Comment by Puneet-here** [comment]
### Proposal 

The add payment method was meant to be at the bottom but after wrapping it with the offlineIndicator its position changed. 

To fix this we have to pass styles.flex1 in style and contentContainerStyle(this prop is not available right now but a PR is up for it) prop at `OfflineWithFeedback` https://github.com/Expensify/App/blob/7b21103bb56eb0cf3fc1561715613d2a16c3255f/src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js#L311

`contentContainerStyle` will be used in the style here
https://github.com/Expensify/App/blob/7b21103bb56eb0cf3fc1561715613d2a16c3255f/src/components/OfflineWithFeedback.js#L92

Now we have to do to something about the margin and padding because we have one more issue, the error for payment method button doesn't look good with the current spacing (check the image below)
Without any changes:

<img width="374" alt="Screenshot 2022-10-08 at 9 35 39 PM" src="https://user-images.githubusercontent.com/101883770/194716747-3884e533-dceb-4b85-a557-2e203ba4d0dd.png">

After correcting the position:
<img width="374" alt="Screenshot 2022-10-08 at 9 32 47 PM" src="https://user-images.githubusercontent.com/101883770/194716625-2b0fe697-f8a8-46e4-ab01-864cba911695.png">

so I propose we only use styles.mb4 when this.props.userWallet.errors contains some value because we don't want the bottom margin when showing error and the error is from userWallet. https://github.com/Expensify/App/blob/7b21103bb56eb0cf3fc1561715613d2a16c3255f/src/pages/settings/Payments/PaymentMethodList.js#L230

Or we can just simply remove the bottom margin from the button and  styles.pv2 from offlineIndicator and add pb4 at the View below https://github.com/Expensify/App/blob/7b21103bb56eb0cf3fc1561715613d2a16c3255f/src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js#L311

After my solution this will be the result -


Without error
<img width="374" alt="Screenshot 2022-10-08 at 9 38 59 PM" src="https://user-images.githubusercontent.com/101883770/194716926-57470b7e-cd5a-40e5-a471-4d870a7d1dda.png">


With error

<img width="374" alt="Screenshot 2022-10-08 at 9 38 44 PM" src="https://user-images.githubusercontent.com/101883770/194716935-0ad9bae3-2b20-4287-94c8-dc341486dc94.png">

**Comment by Puneet-here** [comment]
This is the [PR](https://github.com/Expensify/App/pull/9622).

**Comment by shawnborton** [comment]
I like the idea of moving this button down to the bottom of the screen. 

For the error message, is the error in relation to submitting the form overall, or related to the bank account row that is listed?

**Comment by aldo-expensify** [comment]
Moving the button to the bottom is a good idea IMO, I think we can try to be consistent on that. The only thing I don't like much about it is how that error pushes the button up...

**Comment by shawnborton** [comment]
Well that's why I am trying to understand if the error is attached to submitting the form or the specific bank row. I believe we put overall form errors right above the bottom submit button. But other errors are docked right below the piece of UI that has an error.

**Comment by Puneet-here** [comment]
I can only say that it won't be related to the payment method that we will be listing above (paypal, bank accounts) because we are wrapping them individually so error for specific payment method will be shown below them. This error that we are checking for this button is from `userWallet`. I am not sure what type of errors we get `userWallet` object.

**Comment by shawnborton** [comment]
Got it, so then if the error is related to the overall form submission, I think the error message goes above the bottom-docked green button. We should also update to use the new dot-indicator error message style.

**Comment by Puneet-here** [comment]
I think we should confirm first. 
@nkuoch, sorry for the ping but can you tell us about the error that we are passing here, since you made this change at https://github.com/Expensify/App/pull/11162
https://github.com/Expensify/App/blob/7b21103bb56eb0cf3fc1561715613d2a16c3255f/src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js#L311

We are moving the "add payment method" button to the bottom and we wanna decide where the error should be shown.

**Comment by stephanieelliott** [comment]
Alright to recap, it sounds like we'll move the button to the bottom, now we're just figuring out where the error will be shown relative to the button -- waiting for @nkuoch to provide some more context on the type of error this is so that @Puneet-here can update the proposal he shared in https://github.com/Expensify/App/issues/11414#issuecomment-1272350240. 

I'm OOO til 10/25, reapplied the `External` label to get a new CME/BZ team member! Thanks @slafortune

**Comment by aldo-expensify** [comment]
No overdue, waiting for input

**Comment by aldo-expensify** [comment]
I had a look at the code and it is possible to have errors per payment method and a "user wallet" error.

The payment methods errors appear under each payment method. The "user wallet" error is appearing below the "Add payment method" button. We are considering putting the "user wallet" error above the button. The only downside I see with this is that if it gets close to an account, it could be ambiguous to tell if the error is for the "wallet" (button error) or it is for payment method at the bottom.

![image](https://user-images.githubusercontent.com/87341702/196262622-81b1664c-3706-429e-b4a0-f38843242861.png)

@shawnborton is this information useful to you?


"User wallet error":  notice that it is wapping `<PaymentMethodList>`

https://github.com/Expensify/App/blob/6db559f9b42dbfa5e1e09e60d692272d85039e9c/src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js#L311-L312

Per payment method error: wraps each payment method `<MenuItem>`

https://github.com/Expensify/App/blob/6db559f9b42dbfa5e1e09e60d692272d85039e9c/src/pages/settings/Payments/PaymentMethodList.js#L171-L177

**Comment by shawnborton** [comment]
Cool, that is helpful. And then what kind of error would we see for userWallet.errors? Or rather, when would that happen?

**Comment by aldo-expensify** [comment]
> Cool, that is helpful. And then what kind of error would we see for userWallet.errors? Or rather, when would that happen?

I can see that if you try to set a payment method as the default and you enter the wrong password, you get a `userWallet` error:

![image](https://user-images.githubusercontent.com/87341702/196278806-28c67a18-a2f3-453e-b2c2-b5e94460292c.png)

Here there are other possible errors which could end up there:

https://github.com/Expensify/Web-Expensify/blob/f7653bab8f7aa6b9dd1c2087b37d27dbdf4cdaec/lib/Auth.php#L6431-L6464

**Comment by shawnborton** [comment]
Okay cool, I think that position makes sense then 👍

**Comment by Puneet-here** [comment]
No need to update the proposal, right? We are going to keep the error below the button, correct me if I misinterpreted.

**Comment by slafortune** [comment]
Doubled the payment today

**Comment by aldo-expensify** [comment]
@Puneet-here proposal looks good to me too. I just confirmed with @shawnborton that we wanted to keep the error below the button.

**Comment by arielgreen** [comment]
Hey @Puneet-here @slafortune, can you confirm when the PR will be ready? Let's [keep the pressure on](https://expensify.slack.com/archives/C01SKUP7QR0/p1667064948124109) to get this one closed out!

**Comment by Puneet-here** [comment]
I will raise the PR by tomorrow.

**Comment by aldo-expensify** [comment]
@Puneet-here can you update your branch ([PR](https://github.com/Expensify/App/pull/12502)) with main so we can compile it Android? 🙏

**Comment by Puneet-here** [comment]
> @Puneet-here can you update your branch ([PR](https://github.com/Expensify/App/pull/12502)) with main so we can compile it Android? 🙏

Done!

**Comment by aldo-expensify** [comment]
PR merged

**Comment by Puneet-here** [comment]
@aldo-expensify, I just wanted to know if the compensation will be increased since I also pointed out the error message positioning bug and fixed it which wasn't mentioned in this issue.

**Comment by aldo-expensify** [comment]
@Puneet-here sorry, which error positioning? the error under the "Add payment method" button?

**Comment by Puneet-here** [comment]
Yes this how the error used to show up. since the button had bottom margin and the error had vertical margin there was some extra gap. 
<img width="375" alt="Screenshot 2022-11-09 at 3 34 15 AM" src="https://user-images.githubusercontent.com/101883770/200685579-39b30119-2e15-442f-a456-a9a39a907fc1.png">

**Comment by aldo-expensify** [comment]
@Puneet-here ok, I just noticed that your proposal mentioned this.

@mallenexpensify 👋 can I get your opinion on what @Puneet-here is requesting here: https://github.com/Expensify/App/issues/11414#issuecomment-1307874797

Summary:

- The issue is about moving the button "Add payment method" to the bottom of the screen
- @Puneet-here 's included a small fix for the error's margin when it appears below the button in his [proposal](https://github.com/Expensify/App/issues/11414#issuecomment-1272350240)
- @Puneet-here pointed out that the correction on the margin is not part of the original issue, so he is requesting extra payment for that.

**Comment by slafortune** [comment]
Checking [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1668008095764599)

**Comment by slafortune** [comment]
@Puneet-here  - Yes, we will also compensate with a $250 bonus for that additional fix 👍

**Comment by Puneet-here** [comment]
Thank you🎉

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by slafortune** [comment]
@Puneet-here @rushatgabhane you've been sent the contract on Friday, let me know once you've accepted that and I can get you paid.  👍

**Comment by slafortune** [comment]
Paid @rushatgabhane waiting on @Puneet-here to send payment.

**Comment by Puneet-here** [comment]
Accepted the offer, thanks!
